de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/filesystem/IntelliJProjectImplV2.java

```diff
@@ -67,25 +68,150 @@
+     *                                  or the directory containing the module
+     *                                  file could not be found
      */
     public IntelliJProjectImplV2(@NotNull final Module module) {
         this.module = module;
         this.moduleName = module.getName();
 
+        moduleRoot = getModuleContentRoot(module);
+
+        checkIfContentRootLocatedBelowProjectRoot(module, moduleRoot);
+        checkIfModuleFileLocatedInContentRoot(module, moduleRoot);
+    }
+
+    /**
+     * Returns the content root of the given module.
+     * <p>
+     * This method is used to enforce the current restriction that shared
+     * modules must contain exactly one content root.
+     * </p>
+     *
+     * @param module the module to get the content root for
+     * @return the content root of the given module
+     * @throws IllegalArgumentException if the given module does not have
+     *                                  exactly one content root
+     */
+    @NotNull
+    private static VirtualFile getModuleContentRoot(
+        @NotNull
+            Module module) {
+
         ModuleRootManager moduleRootManager = ModuleRootManager
             .getInstance(module);
 
-        final VirtualFile[] contentRoots = moduleRootManager.getContentRoots();
+        VirtualFile[] contentRoots = moduleRootManager.getContentRoots();
+
+        int numberOfContentRoots = contentRoots.length;
+
+        if (numberOfContentRoots != 1) {
+            throw new IllegalArgumentException(
+                "Modules shared with Saros currently must contain exactly one "
+                    + "content root. The given module " + module + " has "
+                    + numberOfContentRoots + " content roots: " + Arrays
+                    .toString(contentRoots));
+        }
+
+        return contentRoots[0];
+    }
+
+    /**
+     * Checks whether the given content root is located under the project root.
+     *
+     * <p>
+     * This method is used to enforce the current restriction concerning the
+     * module structure.
+     * </p>
+     *
+     * @param module     the module to check the structure for
+     * @param moduleRoot the content root of the given module
+     * @throws IllegalArgumentException if the given content root is not located
+     *                                  under the project root
+     * @throws IllegalStateException    if the project basedir could not be
+     *                                  found
+     */
+    private static void checkIfContentRootLocatedBelowProjectRoot(
+        @NotNull
+            Module module,
+        @NotNull
+            VirtualFile moduleRoot) {
+
+        Project project = module.getProject();
+
+        VirtualFile projectRoot = project.getBaseDir();
+
+        if (projectRoot == null) {
+            throw new IllegalStateException(
+                "The base dir for the project " + project
+                    + " could not be found.");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal " could not be found." 3 times. ([squid:S1192](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1192))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=240&issues=AWYbBs6fvOWp7Yd63TpU&open=AWYbBs6fvOWp7Yd63TpU)

[ISSUE_KEY]: AWYbBs6fvOWp7Yd63TpU